### ISSUE TITLE
Top Level Plan: 1.0 Release!

### ISSUE THREAD

**Issue by markmandel** [googleforgames/agones#732]
This is a top level ticket for planning out what is required to be in a 1.0 release, keeping in mind that we want to release 1.0 at some point this year.

This is a sacrificial draft of how to manage the process. Please feel free to provide alternate opinions.

Breaking this down into two areas:

## Timeliine

Goal is to hit 1.0 in 2019. The earlier, the better.

## Functionality

I think the easiest way to track what we feel should be in the 1.0 release should be added as an item in the next upcoming [milestone release](https://github.com/GoogleCloudPlatform/agones/milestones). If a issue should be added / removed - we can discuss it individually on each ticket.

There is is already a set of issues targeting the next release, for a sacrificial draft of what should be included.

I want to make sure we have a clear plan, but also flexibility to add extra features if we find things that are missing. 

## Hardening / Scalability testing

TL;DR: Lots and lots of load tests, let's see where it breaks. Fix it once it breaks. Repeat.

Much of this is covered in [Load testing](https://github.com/GoogleCloudPlatform/agones/issues/412) ticket.

Question at this point: What are our target goals for gameserver fleet / kubernetes size? What do we determine is "success"?

@jkowalski , @ilkercelikyilmaz , @pm7h - I figure you have thoughts here, since you are doing most of this work.

**Comment by roberthbailey** [comment]
It's also reasonable to say that agones is production ready at a smaller scale than we anticipate eventually supporting. As long as we are clear about what scale is supported (and provide api compatibility) we can make continuous performance and scalability improvements past the 1.0 milestone.

**Comment by markmandel** [comment]
> It's also reasonable to say that agones is production ready at a smaller scale than we anticipate eventually supporting. As long as we are clear about what scale is supported (and provide api compatibility) we can make continuous performance and scalability improvements past the 1.0 milestone.

100% agree. So I would rewrite the above to be:

_Once 1.0 features are complete, performance testing will be completed, and the project will publish supported cluster sizes, fleet sizes and throughput metrics based on the current code base._

Works for me. Work for everyone else?

**Comment by markmandel** [comment]
Added a note about schedule, since it keeps coming up

**Comment by markmandel** [comment]
## 0.11 :arrow_right: 0.12
- Ideally all breaking changes and features will be implemented before the 0.12 release
- This essentially becomes a release candidate, and we go into feature freeze at this point.

## 0.12 :arrow_right:  0.13
- Time between 0.12 and 0.13 would then be spent writing more tests, docs, performance testing, etc

## 0.13
- This would be the actual 1.0 release.

## Risks
- We don't actually get all the functionality in in time for 0.12
- There is missing functionality in some of the SDKs, some of which we will need to source technical capability for
- There are outstanding issues around how to test Unity and/or Unreal because licencing.
- Load testing. Unknown on how we want to do this???
- With all the breaking changes, do we need to document/determine upgrade paths

**Comment by roberthbailey** [comment]
sgtm

**Comment by markmandel** [comment]
:star2: CLOSING :star2:
